std/haxe/Rest.hx

```diff
@@ -6,10 +6,23 @@ import haxe.iterators.RestKeyValueIterator;
 private typedef NativeRest<T> = Array<T>;
 
 /**
-	A special type that represents "rest" function argument.
+	A special type that represents a "rest" function argument.
+	
+	The special `...` syntax can be used for convenience and improved readability:
 
-	Should be used as a type for the last argument of a method, representing that
-	arbitrary number of arguments of given type can be passed to that method.
+	```haxe
+	function f(...rest:Int) {
```

Comment:
is it just me or using element type like `:Int` here feels weird?..

Reply:
Not sure what you mean, that's the whole point of the `...` syntax?

Reply:
not sure, i think every language i saw so far use the array type for the actual rest arg local var. as i said, maybe it's just a habit, but to me this reads like rest is an `Int`, but it's actually not, I don't know...

Reply:
Well, at least Java and Kotlin do it like this. I hadn't ever really noticed that TS and C# do it differently, probably because I use varargs so rarely...

Reply:
Oh well ^^

Reply:
I really don't understand why we support `...` in function definition, `function f(...rest:Int) {}` isn't clear at all imo

Reply:
It's not too late to cut it out :)
To me it looks consistent with allowing `call(...collection)` at call site.

Reply:
Kotlin seems even weirder in that regard (`fun f(vararg rest:Int)` and then `f(*args)`) :)